### DISCUSSION TITLE
Feature Request: Let PD create, edit and resize PNGs and JPGs higher than 32760 pixel

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#514] [Ideas]
As I found out the maximum size (for *.jpg/*.png) images creatable (Canvas Size) by PhotoDemon is 32760pixels (height).

Since I get occasionally webpage screenshots and WhatsApp Chat snapshots which are longer 
it is necessary that I can work (edit) with them and append some more stuff at the bottom end.

So I would like to request to increase in the next PD release the maximum allowed length (height) to a limit far above (e.g. 100000 pixel) the current.

Other image editors (like e.g. paint.net) can handle these dimensions but I prefer to stick with PhotoDemon because of other handling features.

Thank you
Peter Steiner

**Comment by hi5** [comment]
This might be related https://github.com/tannerhelland/PhotoDemon/issues/413 and thus part of the deal when using PD.

Edit: this is just an idea, but I join images side by side or one beneath the other (or build a layered tiff) using [ImageMagick](https://imagemagick.org/) - actually a fork called [GraphicsMagick](http://www.graphicsmagick.org/) - but the idea is I create or edit the "to be appended part" and use a simple batch file to join the two. This may not be feasibly if you actually need to edit the original larger image but if you simply want to "append" this might be something to consider. Splitting the file can be done too, so they can be edited in PD and then merged back into one. Just thinking outside the box.

**Reply by pstein** [reply]
@hi5:

I don't think this is #413 related. Its not an "out of memory" problem.
It is just an unnecessary upper limit of height (and width) which can be changed most probably by just change the hard coded threshold in source code.

Appending stuff at the bottom is only a minor part of that.
Mostly I want to move selected rectangle parts of the image inside the bigger image canvas.

**Reply by hi5** [reply]
It is indeed hard coded, but it seems that it is the upper limit in vb6:

> You cannot set a vb6 integer to 40000 as they are signed 16 bit numbers so +32767 is the maximum.
> -- https://stackoverflow.com/questions/5895816/vb6-overflow-error-with-large-integers

But the developer will be able to confirm.

**Reply by pstein** [reply]
@hi5:

Maybe its a limitation of VB6. But its not a limitation of official PNG or JPG format definition.
So the developer just has to switch from VB6 data type "integer" to data type "long".
In current times when RAM space becomes neglectable this should be done easily.

**Comment by tannerhelland** [comment]
Hello @pstein .  Thank you for this suggestion.  (And thank you to @hi5 for many thoughtful responses!)

The latest PhotoDemon nightly build now allows for resizing an image (or canvas) in up to 100,000 pixels in either dimension.  The old limit was indeed a legacy value created many years ago, and it was due for a revisit.

That said, note that PhotoDemon is a 32-bit application, so it has a permanent 4 GB limit on application memory.  This means that edited images should always stay below ~400 million pixels total (and ideally about half that) to avoid running out of memory.  The actual image size limit varies depending on available system resources, the number of layers in an image, and what actions you intend to apply (e.g. some complex filters require multiple image copies in the background, and extremely large images may run out of memory).

I still have work to do to validate this new 100,000 pixel limit against all the various image formats PhotoDemon supports.  Some formats have dimension limits well below 100k (for example, JPEGs are limited to 65k pixels in a single direction) and I have not yet updated *every* image export function to validate image size.  This will take some time.

Regardless, please let me know if you encounter any unexpected trouble with the new image size limit.  I have only done some quick testing and things looked OK, but more testing is needed to make sure this hasn't caused any weird problems elsewhere in the project.

Thank you!
